### PR TITLE
docs: add replay guide & concept

### PR DESCRIPTION
**Pull request by arinda-arif**
Added documentation on how to use replay feature and the concept of replay. Will add more doc on this part after added backup in replay.

### CODE REVIEW

File: docs/docs/concepts/overview.md [hunk 1]
```diff
@@ -415,7 +415,21 @@ At the moment, Optimus supports BigQuery datastore for 3 types of resources:
 TODO
 
 ## Replay & Backups
-TODO
+A job might need to be re-run (backfill) due to business requirement changes or other various reasons. Optimus provides 
+an easy way to do this using Replay feature. Replay accepts which job and range of date to be updated, validating it,
```
Comment (kushsharma) [inline docs/docs/concepts/overview.md:419]: `way to do this using Replay.`

File: docs/docs/concepts/overview.md [hunk 2]
```diff
@@ -415,7 +415,21 @@ At the moment, Optimus supports BigQuery datastore for 3 types of resources:
 TODO
 
 ## Replay & Backups
-TODO
+A job might need to be re-run (backfill) due to business requirement changes or other various reasons. Optimus provides 
+an easy way to do this using Replay feature. Replay accepts which job and range of date to be updated, validating it, 
+and clearing the tasks in the scheduler, making the tasks available to be picked up by the scheduler and got re-run.
```
Comment (kushsharma) [inline docs/docs/concepts/overview.md:420]: picked up by the scheduler and `for` re-run.

File: docs/docs/concepts/overview.md [hunk 3]
```diff
@@ -415,7 +415,21 @@ At the moment, Optimus supports BigQuery datastore for 3 types of resources:
...
 ## Replay & Backups
-TODO
+A job might need to be re-run (backfill) due to business requirement changes or other various reasons. Optimus provides 
+an easy way to do this using Replay feature. Replay accepts which job and range of date to be updated, validating it, 
+and clearing the tasks in the scheduler, making the tasks available to be picked up by the scheduler and got re-run. 
+Replay will also figure the dependents of the job and re-run all the downstream jobs within the same project just after 
+the upstream tasks are finished.
+
+When validating, Optimus checks if replay with same job and date are currently running and also check if the task 
+scheduler instances are still running to avoid any duplication and confusion. Optimus provides several replay workers so
```
Comment (kushsharma) [inline docs/docs/concepts/overview.md:425]: avoid any duplication and `conflicts`.

File: docs/docs/concepts/overview.md [hunk 4]
```diff
@@ -415,7 +415,21 @@ At the moment, Optimus supports BigQuery datastore for 3 types of resources:
...
-TODO
+A job might need to be re-run (backfill) due to business requirement changes or other various reasons. Optimus provides 
+an easy way to do this using Replay feature. Replay accepts which job and range of date to be updated, validating it, 
+and clearing the tasks in the scheduler, making the tasks available to be picked up by the scheduler and got re-run. 
+Replay will also figure the dependents of the job and re-run all the downstream jobs within the same project just after 
+the upstream tasks are finished.
+
+When validating, Optimus checks if replay with same job and date are currently running and also check if the task 
+scheduler instances are still running to avoid any duplication and confusion. Optimus provides several replay workers so 
+several replays can be run in parallel.
```
Comment (kushsharma) [inline docs/docs/concepts/overview.md:426]: we don't need to mention that replay has several workers, this just causes confusion about what these workers are.

File: docs/docs/concepts/overview.md [hunk 5]
```diff
@@ -415,7 +415,21 @@ At the moment, Optimus supports BigQuery datastore for 3 types of resources:
...
+
+After the replay worker cleared the task instances in the scheduler, the worker finished its job and let the scheduler do
```
Comment (kushsharma) [inline docs/docs/concepts/overview.md:428]: People don't know what replay worker is. 
"the worker finished its job", what worker, what job?

File: docs/docs/concepts/overview.md [hunk 6]
```diff
@@ -415,7 +415,21 @@ At the moment, Optimus supports BigQuery datastore for 3 types of resources:
...
+Replay will also figure the dependents of the job and re-run all the downstream jobs within the same project just after 
+the upstream tasks are finished.
+
+When validating, Optimus checks if replay with same job and date are currently running and also check if the task 
+scheduler instances are still running to avoid any duplication and confusion. Optimus provides several replay workers so 
+several replays can be run in parallel.
+
+After the replay worker cleared the task instances in the scheduler, the worker finished its job and let the scheduler do 
+the rest. However, Optimus provides a replay syncer to frequently check the status in the scheduler (every 5 minutes), so 
+the latest state of replay will always be updated.
```
Comment (kushsharma) [inline docs/docs/concepts/overview.md:430]: `latest state of replay` compares to what? We need to describe this a little more.

File: docs/docs/guides/replay.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+## Run a replay
+
+In order to run a replay, run the following command
+
+```
+$ optimus replay run sample-job 2021-01-01 2021-02-01 --project sample-project --namespace sample-namespace
+```
+
+Replay accepts three arguments, first is DAG name (required) that used in optimus specification, second is
```
Comment (kushsharma) [inline docs/docs/guides/replay.md:18]: `DAG name`

File: docs/docs/guides/replay.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+You will see the latest replay status including the status of each run of your replay.
+
+
+## Get list of replays
+
+List of recent replay of a project can be checked using this sub command:
+
+```
```
Comment (kushsharma) [inline docs/docs/guides/replay.md:43]: use three backticks and then `shell` for language-based formatting here like ```shell

File: docs/docs/concepts/overview.md (hunk 2 above)
Comment (arinda-arif) [reply docs/docs/concepts/overview.md]: removed this part as it is mentioned in the next paragraph.

File: docs/docs/concepts/overview.md (hunk 6 above)
Comment (arinda-arif) [reply docs/docs/concepts/overview.md]: rephrased this

**Comment by arinda-arif** [comment]
Hi @kushsharma, I have fixed the incorrect grammar as suggested.

**Comment by kushsharma** [comment]
@arinda-arif docs look good, you also need to add a new heading in `sidebar.js` to make this page visible in docs.

**Comment by arinda-arif** [comment]
added to sidebar @kushsharma, please check 😃

**Pull request merged**